### DISCUSSION TITLE
A simple user configuration to customize OneDarkPro

### DISCUSSION THREAD

**Discussion by maxmx03** [olimorris/onedarkpro.nvim#182] [Ideas]
The current color scheme utilizes highlights and styles for customization, but it has certain limitations. For example, if you want to change the CmpKind highlight group to appear in reverse, the colors are removed because your configuration overrides the entire highlight group.

My suggestion is to update the highlight group without completely overriding everything. This approach allows you to retain the existing configuration and append it with the new one.

```lua
function get_highlight_group(name)
    local highlight_group = vim.api.nvim_get_hl(0, { name = name, link = false })

    if vim.tbl_isempty(highlight_group) then
        return
    end

    return highlight_group
end

function update_highlight_group(highlights)
    if vim.tbl_isempty(highlights) then
        return
    end

     for highlight_name, highlight_val in pairs(highlights) do
        local highlight_group = get_highlight_group(highlight_name)

        if type(highlight_group) == 'table' then
            local val = vim.tbl_extend('force', highlight_group, highlight_val)

            vim.api.nvim_set_hl(0, highlight_name, val)
        end
    end
end
```

Now, you can use the update_highlight_group function to modify the OneDarkPro theme:

```lua
require("onedarkpro").setup({
  highlights = {
    ['@variable'] = { italic = false, bold = true },
    CmpItemKindText  = { reverse = true },
  }
```

With this new configuration, the highlight group will not be entirely overridden. You can add or modify specific properties while preserving the existing configuration.

**Comment by olimorris** [comment]
Hey @maxmx03! Thanks for your thoughts and I would be more than happy to accept a PR for this. `reverse` is something I'd not tested or given much thought to but think it's a really useful addition.
